Fix #1350: Topic activity back navigation icon

rt4914 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

This PR replaces the `hamburger` icon implementation in `TopicActivity` to `back-arrow` icon.

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Commit:
Back navigation button added

Commit:
Updated test cases

veena14cs commented on app/src/main/java/org/oppia/app/topic/TopicViewModel.kt:
```diff
@@ -0,0 +1,58 @@
+package org.oppia.app.topic
+
+import androidx.fragment.app.Fragment
```
Remove this unused import.

rt4914 replied:
Done.

veena14cs reviewed (changes requested):
Test case is missing for back navigation icon.

Commit:
Updated test cases

rt4914 commented:
> Test case is missing for back navigation icon.

I have added test cases and shifted them to `TopicFragmentTest` and removed `TopicActivityTest` as we do not need that file.

veena14cs reviewed (approved):
LGTM.

Commit:
Merge remote-tracking branch 'upstream/develop' into topic-activity-back-navigation-icon

Commit:
Merge remote-tracking branch 'upstream/develop' into topic-activity-back-navigation-icon
